web/containers/admin/AssessmentPanel/index.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { firebaseConnect, firestoreConnect, isLoaded } from 'react-redux-firebase';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import AssessmentPanel from '../../../components/admin/AssessmentPanel';
+
+class AssessmentPanelContainer extends React.Component {
+  static contextTypes = {
+    store: PropTypes.object.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // one of 'hacker', 'mentor', 'volunteer'
+      applicantType: 'hacker',
+      // all documents from <applicantType>_main_info collection
+      applicants: {},
+      // single document from <applicantType>_main_info for selected applicant
+      selectedApplicantMainInfo: {},
+      // single document from <applicantType>_long_info collection for selected applicant
+      selectedApplicantLongInfo: {},
+    };
+  }
+
+  componentWillMount() {
+    this.setState({ applicants: this.getApplicants() });
+  }
+
+  // returns all documents from <applicantType>_main_info
+  getApplicants() {
+    const {
+      firestore,
+    } = this.props;
+
+    const { applicantType } = this.state;
+    const { hackerMainInfo, volunteerMainInfo, mentorMainInfo } = firestore.data;
+    const applicants = {
+      hacker: hackerMainInfo,
+      volunteer: volunteerMainInfo,
+      mentor: mentorMainInfo,
+    };
+    return applicants[applicantType];
+  }
+
+  // promise which returns <applicant_type>_long_info document for a particular applicant
+  getLongInfo(applicantId) {
+    const { applicantType } = this.state;
+    // http://docs.react-redux-firebase.com/history/v2.0.0/docs/firestore.html#manual
+    const { store } = this.context;
+    const { firestore } = store;
+
+    const longInfoCollections = {
+      hacker: 'hacker_long_info',
+      volunteer: 'volunteer_long_info',
+      mentor: 'mentor_long_info',
+    };
+
+    return firestore.get(
```

Comment:
I've been noticing delays between clicking on applicants and displaying the info. 

I'm suspecting that `firestore.get` makes a new query every time without checking the cache, but let me know if I'm wrong.

One way to get around it is checking `this.props.firestore.data.hacker_long_info.[id]` and checking if we've already queried the id before. Then we don't need to call get.

But I think a more reliable/preferred way is to query "automatically" in whatever component is loading the long info. Then it would "automatically" load it using the best approach (by checking the cache/setting listeners, etc).

```
compose(
  firestoreConnect(({ props }) => {
    // get collection name from props, hacker/volunteer
    // get id from props
    return { collection: <collection name>, doc: <"id"> }
  }),
```

Reply:
You're right, `firestore.get` doesn't do any of the automatic stuff that `firestoreConnect` does. 

I went with the second approach you suggested, added `ApplicantInfoPanel` which accepts the `applicantId` and `applicantType` and checks the store / adds listeners if needed for `longInfo` and `shortInfo` for that applicant.